File: Quantum.School.AngularClient/ClientApp/src/app/models/class-schedule.ts
import { BaseModel } from './base-model';
import { Subject } from './subject';
import { Teacher } from './teacher';
import { Student } from './student';

export interface ClassSchedule extends BaseModel
{
	location: string;

	subject: Subject;

	teacher: Teacher;

	students: Student[];
}

export interface ClassScheduleEntry
{
	location: string;

	subject: string;

	teacher: string;

	students: Student[];
}

File: Quantum.School.Core/Models/ClassSchedule.cs
using System;
using System.Collections.Generic;

namespace Quantum.School.Core.Models
{
	public class ClassSchedule : Entity<Guid>
	{
		public virtual Subject Subject { get; set; }

		public virtual Teacher Teacher { get; set; }

		public string Location { get; set; }

		public virtual ICollection<Student> Students { get; set; }
	}
}
